cypress/integration/focus-trap-demo.spec.js

```diff
@@ -214,19 +214,96 @@ describe('focus-trap', () => {
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'activate inner trap' })
+        .as('lastlyFocusedElInOuterTrap')
+        .click();
+
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'nothing' })
+        .should('be.focused')
+        .as('focusedElInInnerTrap');
+
+      verifyCrucialFocusTrapOnClicking('@focusedElInInnerTrap');
+
+      // only element in the innermost active trap can be focused
+      cy.get('@firstTabbableElInOuterTrap').click().should('not.be.focused');
+
+      // deactivate inner trap and outer trap element can be focused again
+      cy.findByRole('button', {
+        name: 'deactivate and close inner trap',
+      }).click();
+      cy.get('@lastlyFocusedElInOuterTrap').should('be.focused');
+
+      // deactivate outer trap and element outside of trap can be focused again
+      cy.get('@firstTabbableElInOuterTrap').click();
+      cy.get('@lastlyFocusedElBeforeTrapIsActivated').should('be.focused');
+    });
   });
 
   describe('demo: sibling', () => {
     // TODO
   });
 
   describe('demo: tif', () => {
-    // TODO
+    it(`when trap is activated, if there is not any tabbable element in the trap, focus-trap will try to focus the element specified by option "fallbackFocus"`, () => {
+      cy.get('#demo-tif').as('testRoot');
+
+      // activate trap(no tabbable element inside) and the container element(which is the fallback element specified) should be focused
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'activate trap' })
+        .as('activate')
+        .click();
+      cy.get('@testRoot').get('#tif').should('be.focused');
+      verifyCrucialFocusTrapOnClicking('#tif');
+
+      // deactivate trap
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'deactivate trap' })
+        .click();
+
+      // activate trap(tabbable element inside) and the first tabbable element should be focused;
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'show focusable button' })
+        .click();
+      cy.get('@activate').click();
+      cy.get('@testRoot')
+        .findByRole('button', { name: 'hide focusable button' })
+        .as('firstTabbableElInOuterTrap')
+        .should('be.focused');
+      verifyCrucialFocusTrapOnClicking('@firstTabbableElInOuterTrap');
+    });
   });
 
   describe('demo: input', () => {
-    // TODO
+    it(`if current focused element is already in the trap, focus activation does not change its selection range"`, () => {
+      cy.get('#demo-tif').as('testRoot');
+
+      // trap is activated after input change and input selection range is not changed
+      cy.get('@testRoot').get('#focused-input8').as('inputElInTrap').type('1');
+      verifyCrucialFocusTrapOnClicking('@inputElInTrap');
+
+      cy.get('@inputElInTrap').then(([input]) => {
+        expect(input.selectionStart).to.equal('1'.length);
+        expect(input.selectionEnd).to.equal('1'.length);
+      });
+    });
```

Comment:
As with previous tests, I'm thinking we should follow the pattern we established and deactivate the trap, and then test that the input still has focus (since it would've also been the element that had focus when the trap was activated).

Reply:
Sounds an essential focus trap check, will add it